src/extension.ts

```diff
@@ -584,6 +602,39 @@ function launchMetals(
     } else {
       outputChannel.appendLine("Debugging Scala sources is not supported");
     }
+    if (features.decorationProvider) {
+      client.onNotification(DecorationTypeDidChange.type, options => {
+        decorationType = window.createTextEditorDecorationType(options);
+      });
+      client.onNotification(DecorationsRangesDidChange.type, params => {
+        outputChannel.appendLine(JSON.stringify(params));
+        const uris = window.visibleTextEditors.map(e => e.document.uri);
+        outputChannel.appendLine(JSON.stringify(uris));
+        const editor = window.activeTextEditor;
+        if (editor) {
+          outputChannel.appendLine(
+            JSON.stringify(editor.document.uri.toString())
+          );
+        }
+        if (editor && params.uri == editor.document.uri.toString()) {
+          outputChannel.appendLine("adding decorations");
+          outputChannel.appendLine(JSON.stringify(params.options));
+          const options = params.options.map<DecorationOptions>(o => {
+            return {
+              range: new Range(
+                new Position(o.range.start.line, o.range.start.character),
+                new Position(o.range.end.line, o.range.end.character)
+              ),
+              hoverMessage: o.hoverMessage,
+              renderOptions: o.renderOptions
+            };
+          });
+          editor.setDecorations(decorationType, options);
+        } else {
+          outputChannel.appendLine("no editor");
```

Comment:
This will print `no editor` also in case `params.uri !== editor.document.uri.toString()`, is that intentional?

Reply:
Good point, I updated the message to say that decorations were ignored for a "non-active document".